fraud/sync.go

```diff
@@ -46,11 +48,13 @@ func (f *ProofService) syncFraudProofs(ctx context.Context) {
 			connStatus = e.(event.EvtPeerIdentificationCompleted)
 		}
 
-		// ignore ourselves as a peer
-		if connStatus.Peer == f.host.ID() {
+		// ignore already discovered peers or ourselves as a peer
+		if _, ok := peerCache[connStatus.Peer]; ok || connStatus.Peer == f.host.ID() {
 			i--
 			continue
 		}
+
+		peerCache[connStatus.Peer] = struct{}{}
```

Comment:
We should add the peer to the cache when the request succeeded.

Reply:
I'd suggest to cache regardless the request result.

Reply:
From sync discussion. There is not so much value "from giving a peer a second chance" if we fail to request a peer.

Reply:
So resolving